newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/VerticeRow.js

```diff
@@ -1,66 +1,134 @@
-            type="number"
-            id="vertice-y"
-            onChange={(e, value) =>
-              props.onChangeVerticeY(parseFloat(value || 0))
-            }
-          />
-        </TableRowColumn>
-        <TableRowColumn style={styles.toolColumn}>
-          {!!props.onRemove && (
-            <IconButton
-              size="small"
-              onClick={props.onRemove}
-              disabled={!props.canRemove}
-            >
-              <Delete />
-            </IconButton>
-          )}
-        </TableRowColumn>
-      </TableRow>
-    )}
-  </ThemeConsumer>
-);
+const VerticeRow = ({
+  verticeX,
+  verticeY,
+  parentVerticeId,
+  ...props
+}: Props) => {
+  const DragSourceAndDropTarget = React.useMemo(
+    () => makeDragSourceAndDropTarget(`collision-mask-${parentVerticeId}`),
+    [parentVerticeId]
+  );
+  const [verticeXInputValue, setVerticeXInputValue] = React.useState<string>(
+    verticeX.toString()
+  );
+  const [verticeYInputValue, setVerticeYInputValue] = React.useState<string>(
+    verticeY.toString()
+  );
+  React.useEffect(
+    () => {
+      setVerticeXInputValue(verticeX.toString());
+    },
+    [verticeX]
+  );
+  React.useEffect(
+    () => {
+      setVerticeYInputValue(verticeY.toString());
+    },
+    [verticeY]
+  );
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <DragSourceAndDropTarget
+          beginDrag={() => {
+            props.setDragged();
+            return {};
+          }}
+          canDrag={() => true}
+          canDrop={() => true}
+          drop={() => {
+            props.drop();
+          }}
+        >
+          {({ connectDragSource, connectDropTarget, isOver, canDrop }) =>
+            connectDropTarget(
+              <tr
+                className="MuiTableRow-root"
+                style={{ backgroundColor: muiTheme.list.itemsBackgroundColor }}
+              >
+                {connectDragSource(
+                  <td
+                    style={
+                      isOver
+                        ? { borderTop: `2px solid ${dropIndicatorColor}` }
+                        : undefined
+                    }
+                  >
+                    <DragHandle />
+                  </td>
+                )}
+                <TableRowColumn style={styles.coordinateColumn}>
+                  <TextField
+                    margin="none"
+                    value={verticeXInputValue}
+                    type="number"
+                    id="vertice-x"
+                    onChange={(e, value) => {
+                      setVerticeXInputValue(value);
+                      const valueAsNumber = parseFloat(value);
+                      if (!isNaN(valueAsNumber)) {
+                        props.onChangeVerticeX(valueAsNumber);
+                      }
+                    }}
+                  />
+                </TableRowColumn>
+                <TableRowColumn style={styles.coordinateColumn}>
+                  <TextField
```

Comment:
Maybe a SemiControlledTextField would have worked? Give it a look, it can maybe avoid you to have these 2 states.

In theory, it is made to solve the issue you had: it's is "semi controlled" in the sense that the value is coming from the value prop BUT when the user change the value, the user can enter anything.